evap/staff/importers.py

```diff
@@ -97,28 +98,98 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
+            name_collision = "name_en"
+        if Course.objects.filter(semester=semester, name_de=self.name_de).exists():
+            name_collision = "name_de" if not name_collision else "both"
+        return name_collision
+
+    def find_and_set_existing_course(self, semester, importer):
+        courses = Course.objects.filter(
             semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
         )
-        course.save()
-        course.responsibles.set([responsible_dbobj])
-        course.degrees.set(self.degrees)
-        evaluation = Evaluation(
-            vote_start_datetime=vote_start_datetime,
-            vote_end_date=vote_end_date,
-            course=course,
-            wait_for_grade_upload_before_publishing=self.is_graded,
-        )
-        evaluation.save()
-        evaluation.contributions.create(
-            evaluation=evaluation,
-            contributor=responsible_dbobj,
-            role=Contribution.Role.EDITOR,
-            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-        )
+        if courses.exists():
+            course = courses.first()
+            differing_attributes = []
+            if course.type != self.course_type:
+                differing_attributes.append("course type")
+            if len(course.responsibles.all()) != 1 or course.responsibles.first().email != self.responsible_email:
+                differing_attributes.append("responsible person")
+            if not differing_attributes:
+                evaluations = Evaluation.objects.filter(course=course)
+                if len(evaluations) == 1:
+                    if evaluations.first().wait_for_grade_upload_before_publishing == self.is_graded:
+                        self.existing_course = course
+                        return course
+                    importer.errors[ImporterError.COURSE].append(
+                        _(
+                            "Course {} ({}) does already exist in this semester but the grading of the evaluation does not match."
+                        ).format(self.name_en, self.name_de)
+                    )
+                else:
+                    importer.errors[ImporterError.COURSE].append(
+                        _(
+                            "Course {} ({}) does already exist in this semester and is identical but must have only one evaluation."
+                        ).format(self.name_en, self.name_de)
+                    )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} ({}) does already exist in this semester but the {} {} not match.").format(
+                        self.name_en,
+                        self.name_de,
+                        " and ".join(differing_attributes),
+                        "does" if len(differing_attributes) == 1 else "do",
```

Comment:
This could probably use `ngettext`, see https://docs.djangoproject.com/en/4.0/topics/i18n/translation/#pluralization and the examples already in the codebase